### ISSUE TITLE
rke2-canal

### ISSUE THREAD

**Issue by Buzqi** [rancher/rke2#4491]
I'm installing RKe2 in a fully isolated env, with no internet connection, therefore I'm using RPMS for the install. 

I don't want the rke2 server to install the rke2-canal chart as I will make use of Calico for the CNI. 

I have tried 

disabled: rke2-canal. 

In the config file before starting the server but that does not seem to work. and on top of that is there any way to change the service cidr and pod cidr that comes with rke2 by default? 

Asking because I'm not using kubeadm for the install.

**Comment by rbrtbnfgl** [comment]
`cni: none` should disable the cni installation

**Comment by Buzqi** [comment]
and that should be set in the config.yaml file right?

**Comment by rbrtbnfgl** [comment]
yes

**Comment by Buzqi** [comment]
and for the cluster and service pods ip would

cluster-cidr: x.x.x.x
service-cidr: x.x.x.x 

Ovveride the 10.42.x. & 10.43.x?

**Comment by rbrtbnfgl** [comment]
Yes

**Comment by Buzqi** [comment]
So if my understanding of how everything is packaged is correct. 

Calico configuration and its binaries is already in the image file needed for the install correct?

I'm referring to "rke2-images.linux.amd64.tar"

**Comment by rbrtbnfgl** [comment]
In case you want to use calico the right configuration should be `cni: calico`

**Comment by Buzqi** [comment]
yeah, I had that option at first, However I realized there is a helm job that still attempt to install canal. My goal ideally is to fully get rid of canal and install calico as the cni plugin, I'm no sure how that is done for the rke2 as I have done with with k8s only.  Ik rke2 using helm packages. 
When I added calico it tries to pull the image from the internet. My guess is I should push these images to my local registry right?
Does the the rpm for helm contains all the necessary files to run calico as well?

My config.yaml file look something like this

selinux: true
write-kubeconfig-mode: "0644"
cni: none
disable: 
- rke2-canal
cluster-cidr: "x.x.x.0/16"
service-cidr: "x.x.x.0/16"

**Comment by Buzqi** [comment]
Then a local registry should be stood up for these images right?

**Comment by Buzqi** [comment]
or are they packaged in helm as well?
